### ISSUE TITLE
CPY001: copyright without a year

### ISSUE THREAD

**Issue by nijel** [astral-sh/ruff#10062]
There are good reasons to omit year in copyright notice, see for example https://daniel.haxx.se/blog/2023/01/08/copyright-without-years/, however ruff CPY001 enforces it:

Code:

```py
# Copyright (c) Michal Čihař <michal@weblate.org>
```

Fails:

```console
$ ruff check --preview weblate/wsgi.py 
weblate/wsgi.py:1:1: CPY001 Missing copyright notice at top of file
  |
1 | # Copyright (c) Michal Čihař <michal@weblate.org>
  |  CPY001
```

**Comment by charliermarsh** [comment]
Makes sense.

**Comment by charliermarsh** [comment]
For the record, you can customize the regex here. It's exposed as a setting: https://docs.astral.sh/ruff/settings/#lint_flake8-copyright_notice-rgx.

**Comment by nijel** [comment]
I know, but I would expect that out of the box the most common situations are handled gracefully. Following https://reuse.software/spec/#comment-headers would be really great

**Comment by nijel** [comment]
Here are the regexps reuse-tool uses:

https://github.com/fsfe/reuse-tool/blob/00a8d949b245218541fd0ece8a64c5480a9e84d7/src/reuse/_util.py#L115-L131

**Comment by charliermarsh** [comment]
The reuse link does say: "The copyright notice SHOULD contain the year of publication."

**Comment by nijel** [comment]
Indeed, there is SHOULD and not MUST. A copyright statement without a year is valid, and there can be good reasons for that (see before mentioned https://daniel.haxx.se/blog/2023/01/08/copyright-without-years/ for reasoning from curl developers).

**Comment by charliermarsh** [comment]
That's true, but it seems reasonable for the rule to enforce a SHOULD by default. You can use https://docs.astral.sh/ruff/settings/#lint_flake8-copyright_notice-rgx to change the regex if you'd like.

**Comment by charliermarsh** [comment]
I'm going to leave this as-is given that the spec is a SHOULD, and so we include it by default, but allow users to override it.
